public/app/features/explore/TableContainer.tsx

```diff
@@ -99,6 +99,7 @@ export class TableContainer extends PureComponent<Props> {
             subData={subFrames}
             width={tableWidth}
             height={height}
+            maxHeight={600}
```

Comment:
Feels weird to have both height and maxHeight here. What is the interaction between those two?

Reply:
I mean I am not sure how these should be used. Like if I set height = 500 does it mean it will be 500 unexpanded but if some row expands it will grow without limit? And if maxHeight = 600 it will grow only to 600? What if height > maxHeight? So one thing, I assume we don't really need the maxHeight for explore. The other thing is I wonder if there is a better way to express this cause to me it's a bit confusing.

Like here I would sort off expect something more automatic that `if height <= 3 * headerHeight && subRows === true then heigh += 200` or something like that, basically having a minHeight instead of maxHeight I guess.

Because I feel like there are 2 general cases:
1. you have some container and you need to fit into it like a dashboard panel so you don't want to expand on subRows anyway.
2. you don't have the container and you can expand (like explore) but I don't think in that case you care much about maxHeight (assuming whole page will just scroll)

So maybe even just having `expandable: boolean` property would make sense to me for this.

Reply:
I think the main reason for having set height is the case when you have footer in the table. Then you cannot leave the scroll on the container and you have to set the height of the table to be equal to container otherwise you would not see the footer unless you scroll all the way down. Which I assume we don't want?

I would assume somebody from @grafana/grafana-bi-squad should also chime in on this.

Reply:
Including `maxHeight` for footer display makes sense to me. I'm not sure if in every scenario we would be able to rely on container sizing to constrain the table. I think as long as it's kept limited it's not a bad idea.

I'm not sure about the `expandable` flag as it seems like having sub-rows implies the ability to expand the table as well. The same could be accomplished by setting the `maxHeight` to whatever the initial height is.